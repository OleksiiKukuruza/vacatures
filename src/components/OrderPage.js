import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import ProductList from './ProductList';
import OrderContainer from '../containers/OrderContainer';

const StyledOrderPage = styled.div`
  height: 100%;
  width: 100%;
  display: flex;
`;

const OrderPage = ({ selectedVacancyId }) => (
  <StyledOrderPage>
    <ProductList />
    {selectedVacancyId && <OrderContainer />}
  </StyledOrderPage>
);

OrderPage.propTypes = {
  selectedVacancyId: PropTypes.string
};

export default OrderPage;
