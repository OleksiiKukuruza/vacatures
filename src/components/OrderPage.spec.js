import React from 'react';
import { shallow } from 'enzyme';
import OrderPage from './OrderPage';

const setup = (props) => {
  const wrapper = shallow(<OrderPage {...props} />);

  return {
    wrapper
  };
};

describe('OrderPage', () => {
  it('renders only product list if nothing selected', () => {
    const { wrapper } = setup({ selectedVacancyId: null });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders product list if product selected', () => {
    const { wrapper } = setup({ selectedVacancyId: '1' });
    expect(wrapper).toMatchSnapshot();
  });
});
