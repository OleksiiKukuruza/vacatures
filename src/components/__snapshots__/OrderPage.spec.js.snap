// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderPage renders only product list if nothing selected 1`] = `
<styled.div>
  <ProductList />
</styled.div>
`;

exports[`OrderPage renders product list if product selected 1`] = `
<styled.div>
  <ProductList />
  <Connect(Order) />
</styled.div>
`;
